src/Npgsql/VolatileResourceManager.cs

```diff
@@ -138,6 +143,7 @@ public void Commit(Enlistment enlistment)
 
             try
             {
+                _isRMCall.Value = true;
```

Comment:
I avoid relying on thread id because of asp.net thread agility which can switch threads under heavy loads. That is why I tend to use `AsyncLocal` instead, which is supposed to be preserved even in such case. But here I have put a `ThreadLocal` due to `AsyncLocal` not being available.